_source/_posts/2021-12-20-net-azure-cognitive-services.md

```diff
@@ -0,0 +1,829 @@
+
+{% img blog/net-azure-cognitive-services/edit-profile.jpg alt:"Edit profile form field." width:"800" %}{: .center-image }
+
+## Add Azure Storage
+
+To make your profile picture viewable, we need to persist the image somewhere. We'll use Azure Storage to do this and use a custom attribute in Okta to store a unique identifier to associate with the file in Azure. You'll be able to upload and display a profile picture in the app.
+
+First, we'll create the custom attributes in Okta using the Okta admin dashboard.
+
+Log in to the [Okta admin dashboard](https://developer.okta.com/login/). Then navigate to **Directory** > **Profile Editor**. You should see an item named **User (default)**in the list, which is Okta's default user profile template. Select **User (default)** to open the **Profile Editor**. Press the **+Add Attribute** button.
+
+You'll add two string attributes, `profileImageKey` and `personId`:
+1. Use "Profile Image Key" as the display name and `profileImageKey` as the variable name.
+2. Use "Person Id" as the display name for the second attribute and `personId` as the variable name.
+
+See the image below for the "Profile Image Key" attribute inputs.
+
+{% img blog/net-azure-cognitive-services/custom-attribute.jpg alt:"Custom attribute settings showing display and variable name for 'Profile Image Key'" width:"800" %}{: .center-image }
+
+Create another attribute named `personId`. If you look at the "User (default)" user type attribute list, you should see your two new attributes listed at the bottom.
+
+{% img blog/net-azure-cognitive-services/all-attributes.jpg alt:"User (default) user type list of attributes with the two new attributes for 'Profile Image Key' and 'Person ID'." width:"800" %}{: .center-image }
+
+Next, we'll create Azure Storage. If you don't already have an Azure subscription, [make a free account](https://azure.microsoft.com/free/?WT.mc_id=A261C142F) before you begin. We'll walk through the steps using the Azure Portal, but if you are an Azure pro, feel free to use the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) or [Azure PowerShell](https://docs.microsoft.com/en-us/powershell/azure/?view=azps-7.0.0).
+
+Once you have an Azure subscription, open the [Azure Portal](https://portal.azure.com). Open the menu by pressing the hamburger menu on the left, and select **Storage accounts**. Press **+Create** to create a storage account. In the **Create a storage account** view, create a Resource Group if you don't already have one for test projects. I named my Resource Group "OktaDemo". Enter `oktaprofilepicture` as the storage name, select a region (Azure's default selection is fine here), and select **Standard** performance and **Locally-redundant storage (LRS)** for redundancy. Press the **Review + create** button at the bottom as the default options for the remaining selections are acceptable. Press **Create** to complete creating the storage account.
```

Comment:
```suggestion
Once you have an Azure subscription, open the [Azure Portal](https://portal.azure.com). Open the menu by pressing the *hamburger menu* on the left, and select **Storage accounts**. Press **+Create** to create a storage account. In the **Create a storage account** view, create a Resource Group if you don't already have one for test projects. I named my Resource Group "OktaDemo". Enter `oktaprofilepicture` as the storage name, select a region (Azure's default selection is fine here), and select **Standard** performance and **Locally-redundant storage (LRS)** for redundancy. Press the **Review + create** button at the bottom as the default options for the remaining selections are acceptable. Press **Create** to complete creating the storage account.
```
italicize for anyone left on earth who doesn't know this has nothing to do with beef